### PR TITLE
CB-1939 configure enumeration for sssd

### PR DESCRIPTION
**Pull request by keyki**

### CODE REVIEW

File: core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java [hunk 1]
```diff
@@ -326,6 +326,14 @@ private void saveSssdIpaPillar(Cluster cluster, Map<String, SaltPillarProperties
             sssdConnfig.put("domain", kerberosConfig.getDomain());
             sssdConnfig.put("password", kerberosConfig.getPassword());
             sssdConnfig.put("server", kerberosConfig.getUrl());
+            Map<String, List<String>> serviceLocations = getServiceLocations(cluster, blueprintService.isAmbariBlueprint(cluster.getBlueprint()));
+            // enumeration has performance impacts so it's only enabled if Ranger is installed on the cluster
+            // otherwise the usersync does not work with nss
+            if (CollectionUtils.isEmpty(serviceLocations.get("RANGER_ADMIN"))) {
```
Comment (lajosrodek) [inline core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java]: Or simply `sssdConnfig.put("enumerate", !CollectionUtils.isEmpty(serviceLocations.get("RANGER_ADMIN")));`. 😃

Comment (keyki) [reply core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java]: Right, I'll make the change

File: core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java [hunk 2]
```diff
@@ -458,8 +466,7 @@ private void saveGatewayPillar(GatewayConfig gatewayConfig, Cluster cluster, Map
         gateway.put("kerberos", kerberosConfig != null);
 
         boolean ambariBlueprint = blueprintService.isAmbariBlueprint(cluster.getBlueprint());
-        List<String> serviceNames = ambariBlueprint ? ExposedService.getAllServiceNameForAmbari() : ExposedService.getAllServiceNameForCM();
-        Map<String, List<String>> serviceLocation = componentLocator.getComponentLocation(cluster, serviceNames);
+        Map<String, List<String>> serviceLocation = getServiceLocations(cluster, ambariBlueprint);
```
Comment (lajosrodek) [inline core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java]: Maybe worth renaming the variable to `serviceLocations` to be consistent with the method name (and the change above).

File: core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java (hunk 1 above)
Comment (keyki) [reply core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java]: fixed

File: core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java (hunk 2 above)
Comment (keyki) [reply core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java]: fixed

File: core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java (hunk 1 above)
Comment (lajosrodek) [reply core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java]: Thanks. 😃

Comment (keyki) [reply core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java]: Thanks for the review!

File: core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java (hunk 2 above)
Comment (lajosrodek) [reply core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java]: Perfect.

**Pull request merged**